airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresCdcTargetPosition.java

```diff
@@ -51,14 +51,20 @@ static PostgresCdcTargetPosition targetPosition(final JdbcDatabase database) {
 
   @Override
   public boolean reachedTargetPosition(final JsonNode valueAsJson) {
+    final SnapshotMetadata snapshotMetadata = SnapshotMetadata.valueOf(valueAsJson.get("source").get("snapshot").asText().toUpperCase());
     final PgLsn eventLsn = extractLsn(valueAsJson);
 
-    if (targetLsn.compareTo(eventLsn) > 0) {
+    if (SnapshotMetadata.TRUE == snapshotMetadata) {
```

Comment:
Previously we initially looked at event `lsn` and if it reached target we returned `true` iso snapshotMetadata was `FALSE` or `LAST`.
Can you explain why this change?

Reply:
This change is mainly to denote that we dont trigger shutdown while snapshot is going on. there are 3 possible values for `snapshotMetadata` (`TRUE` , `FALSE` and `LAST`). TRUE means that snapshot is going on. LAST means that the record is the last value of snapshot and FALSE means that we are processing the transaction logs.